### PR TITLE
Add worker tests

### PR DESCRIPTION
**Pull request by BarryThePenguin**
`lib/worker.js` keeps an instance of `Task` in module scope, so calling `startWorker()` and `stopWroker()` updates the same instance.

This was a problem when running tests concurrently, as tests would be interacting with the same `Task` instance from other tests.

I think there are a few ways around this, but I was just looking for some input first

### CODE REVIEW

File: lib/worker.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ let worker;
 module.exports.startWorker = startWorker;
 module.exports.stopWorker = stopWorker;
 
-function startWorker() {
+function startWorker(createWorker) {
```
Comment (BarryThePenguin) [inline lib/worker.js:11]: Allows passing in a `MockTask`. Otherwise creates it's own `Task` instance

File: lib/worker.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	const lintJobs = pMapSeries([
+		lint('foo.js')('// some editor text'),
+		lint('bar.js')('// some more editor text'),
+		lint('baz.js')('// some other editor text')
+	], job => {
+		processMessages(task);
+		return job;
+	});
+
+	processMessages(task);
```
Comment (BarryThePenguin) [inline lib/worker.test.js:67]: This was a bit weird.. because of `limit()`, the lint jobs have to be run sequentially

File: mocks/mock-task.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const privateSpace = new WeakMap();
+
+module.exports.processMessages = function (task) {
+	const {emitter, messages} = privateSpace.get(task);
+
+	for (const {id, ...message} of messages) {
+		emitter.emit(id, message);
+	}
+};
```
Comment (BarryThePenguin) [inline mocks/mock-task.js:11]: This is just a small helper to process messages send to the `MockTask`

File: lib/worker.js (hunk 1 above)
Comment (sindresorhus) [inline lib/worker.js:11]:
```suggestion
function startWorker(createWorker = createWorker()) {
```

File: mocks/mock-task.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const {Emitter} = require('event-kit');
+
+const privateSpace = new WeakMap();
+
+module.exports.processMessages = function (task) {
```
Comment (sindresorhus) [inline mocks/mock-task.js:5]:
```suggestion
module.exports.processMessages = task => {
```

File: lib/worker.js (hunk 1 above)
Comment (BarryThePenguin) [reply lib/worker.js:11]: I'm not quite sure how this would work? I didn't think default parameters could reference themselves

Comment (sindresorhus) [reply lib/worker.js:11]: Yeah, it makes no sense. I intended to make it:
```suggestion
function startWorker(createWorker = new Task(require.resolve('./xo-helper.js')) {
```

But thinking more about it, even that is not as readable.

**Comment by BarryThePenguin** [comment]
I think that's about it, probably ok to do a new release now

**Comment by sindresorhus** [comment]
Done :)

**Pull request merged**